src/codegen/generators/expression_generator.rs

```diff
@@ -1008,6 +1015,135 @@ impl<'a, 'b> ExpressionCodeGenerator<'a, 'b> {
             })
     }
 
+    /// generates the result of an pointer binary-expression
+    ///
+    /// - `operator` the binary operator
+    /// - `left` the left side of the binary expression, needs to be an pointer/int-value
+    /// - `left_type` DataTypeInformation of the left side
+    /// - `right` the right side of the binary expression, needs to be an pointer/int-value
+    /// - `right_type` DataTypeInformation of the right side
+    /// - `expression` the binary expression
+    pub fn create_llvm_binary_expression_for_pointer(
+        &self,
+        operator: &Operator,
+        left: &AstStatement,
+        left_type: &DataTypeInformation,
+        right: &AstStatement,
+        right_type: &DataTypeInformation,
+        expression: &AstStatement,
+    ) -> Result<BasicValueEnum<'a>, Diagnostic> {
+        let left_expr = self.generate_expression(left)?;
+        let right_expr = self.generate_expression(right)?;
+
+        let result = match operator {
+            Operator::Plus | Operator::Minus => {
```

Comment:
I think the Minus is behaving like a plus here, you need to negate the index before proceeding here.
Could you add a - case to the correctness test?